File: 0136-single-number/0136-single-number.java
class Solution {
    public int singleNumber(int[] nums) {
        Map<Integer,Integer> map = new HashMap<>();
        for(int i=0; i<nums.length;i++){
            map.put(nums[i],map.getOrDefault(nums[i],0)+1);
        }
        
        for(Integer tx : map.keySet()){
            if(map.get(tx)==1) return tx;
        }
        return 0;
    }
}

File: 0136-single-number/0136-single-number.kt
class Solution {
    fun singleNumber(nums: IntArray): Int {
        val map = HashMap<Int,Int>()
        for(i in nums.indices){
            map[nums[i]] = map.getOrDefault(nums[i],0)+1
        }
        for((key,value) in map){
            if(value==1) return key;
        }
        return 0
    }
}